notebooks/03.02. Supervised Learning.Rmd

```diff
@@ -22,118 +27,177 @@ We will use the following classification algorithms:
 We will use the following metrics to evaluate the models:
  - Confusion Matrix.
  - ROC and AUC Multiclass Curves.
- - Lift Curves.
  
-## Onehot Dataset
-First, we will load the `onehot_train.csv` dataset and apply the classification algorithms to it.
+## Functions
+
+We will define some functions to help us evaluate the models.
+
+First, we will load the libraries we will use.
 
 ```{r}
-# Load the dataset
 library(tidyverse)
 library(caret)
 library(pROC)
 library(ggplot2)
 library(xgboost)
+```
 
-onehot_train <- read.csv("../data/processed/train/onehot_train.csv")
-head(onehot_train)
+The following function will be used to apply the classification algorithms to the datasets. It will train the model, save it, predict the test data, save the predictions, and return the trained model.
+
+```{r}
+apply_algorithm <- function(algorithm, tuneGrid, train_data, test_data, model_path, predictions_path) {
+  # Record start time
+  start_time <- Sys.time()
+  
+  # Convert the target column to a factor
+  train_data$damage_grade <- as.factor(train_data$damage_grade)
+  levels(train_data$damage_grade) <- make.names(levels(train_data$damage_grade))
+  
+  # Train the model
+  set.seed(123)
+  model <- train(
+    x = train_data %>% select(-damage_grade),
+    y = train_data$damage_grade,
+    method = algorithm,
+    trControl = trainControl(method = "cv", number = 2, verboseIter = TRUE, classProbs = TRUE),
+    # trControl = trainControl(method = "cv", number = 5, verboseIter = TRUE, classProbs = TRUE),
```

Comment:
With `k=5`, these are the times for each algorithm:
- `Ranger` -> 3:20
- `Xgbtree` -> 2:47
- `Sgb` -> 53s